README_client.rst

```diff
@@ -0,0 +1,480 @@
+===========================================
+[Beta] Client interface for Scrapinghub API
+===========================================
+
+
+The ``scrapinghub.ScrapinghubClient`` is a new Python client for communicating
+with the `Scrapinghub API`_. It takes best from ``scrapinghub.Connection`` and
+``scrapinghub.HubstorageClient`` and combines it under single interface.
+
+
+.. contents:: :depth: 2
+
+
+Testing
+=======
+
+The client is covered with integration tests based on `VCR.py library`_: there
+are recorded cassettes files in ``tests/client/cassettes`` used instead of HTTP
+requests to real services, it helps to simplify and speed up development.
+
+By default, tests use VCR.py ``once`` mode to:
+
+- replay previously recorded interactions.
+- record new interactions if there is no cassette file.
+- cause an error to be raised for new requests if there is a cassette file.
+
+It means that if you add new integration tests and run all tests as usual,
+only new cassettes will be created, all existing cassettes will stay unmodified.
+
+To ignore existing cassettes and use real service, please provide a flag::
+
+    py.test --ignore-cassettes
+
+If you want to update/recreate all the cassettes from scratch, please use::
+
+    py.test --update-cassettes
+
+Note that internally the above command erases the whole folder with cassettes.
+
+
+Basic usage
+===========
+
+First, you connect to Scrapinghub::
+
+    >>> from scrapinghub import ScrapinghubClient
+    >>> client = ScrapinghubClient('APIKEY')
+    >>> client
+    <scrapinghub.client.ScrapinghubClient at 0x1047af2e8>
+
+Client instance has ``projects`` field for access to client projects.
+
+Projects
+--------
+
+You can list the projects available to your account::
+
+    >>> client.projects.list()
+    [123, 456]
+
+Or check the projects summary::
+
+    >>> client.projects.summary()
+    [{'finished': 674,
+      'has_capacity': True,
+      'pending': 0,
+      'project': 123,
+      'running': 1},
+     {'finished': 33079,
+      'has_capacity': True,
+      'pending': 0,
+      'project': 456,
+      'running': 2}]
+
+And select a particular project to work with::
+
+    >>> project = client.get_project(123)
+    >>> project
+    <scrapinghub.client.Project at 0x106cdd6a0>
+    >>> project.id
+    123
+
+The above is a shortcut for ``client.projects.get(123)``.
+
+Project
+-------
+
+Project instance has ``jobs`` field to work with the project jobs.
+
+Jobs instance is described well in ``Jobs`` section below.
+
+For example, to schedule a spider run (it returns a job object)::
+
+    >>> project.jobs.schedule('spider1', arg1='val1')
+    <scrapinghub.client.Job at 0x106ee12e8>>
```

Comment:
@vshlapakov if we decide to add spider_args later that would be a backwards incompatible change if someone accidentally has spider argument `spider_args`. another problem is that parameters like `units` or `priority` or `spider_settings` are not distinguishable from any other spider argument. by having separate kwarg for spider args we can guarantee that if we add new parameters to schedule endpoint they would not clash with spider args that clients use